lib/web/app/transport.go

```diff
@@ -98,17 +102,20 @@ func newTransport(c *transportConfig) (*transport, error) {
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
-	tr.DialContext = dialFunc(c)
-	tr.TLSClientConfig, err = configureTLS(c)
+
+	t := &transport{
+		c:  c,
+		tr: tr,
+	}
+
+	t.clientTLSConfig, err = configureTLS(c)
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
 
-	return &transport{
-		c:      c,
-		tr:     tr,
-		dialer: websocketsDialer(tr),
-	}, nil
+	tr.DialContext = t.DialContext
+	tr.TLSClientConfig = t.clientTLSConfig
+	return t, nil
```

Comment:
Here you make some changes to `tr`, but then return `t`? Looks like a typo.

Reply:
Not really, the `tr` is inside the `t` (which is the `*transport` we want to return). I will change the order of the changes to make it more clear.